src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -52,6 +52,22 @@ public FileTextLoaderOptionsProvider()
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     public class FileTextLoader : TextLoader
     {
+        /// <summary>
+        /// Guards access to <see cref="_workspace"/> and <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private readonly object _gate = new object();
+
+        /// <summary>
+        /// This is the workspace used to initialize <see cref="_lazyTextAndVersion"/>.
+        /// </summary>
+        private Workspace? _workspace;
+
+        /// <summary>
+        /// A lazy loader for the <see cref="TextAndVersion"/>. The value computed by this is not cached, so the same
+        /// instance can service load requests until the workspace changes.
```

Comment:
👍

Reply:
might be valuable to say that the beneficial purpose of this is primarily to ensure that if we get multiple simultaneous callers that tehy can all depend on the same computation.

Reply:
I rewrote this comment. Let me know if the new text is fine so I can rebase.